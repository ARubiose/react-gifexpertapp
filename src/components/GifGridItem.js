import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ( { id, tittle, url } ) => {
    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={ url } alt={ tittle } />
            <p>{ tittle }</p>
        </div>
    )
}

GifGridItem.propTypes = {
    tittle: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
