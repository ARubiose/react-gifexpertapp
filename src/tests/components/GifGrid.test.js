import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe( 'Testing GifGrid', () => {

    const category = 'One Punch';

    test( 'should correctly show the component', () => {
        
        useFetchGifs.mockReturnValue({
            data: [], 
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/>);
        expect( wrapper ).toMatchSnapshot();

    });

    test('should show items when images are loaded from useFetchGifs', () => {
        
        const gifs = [{
            id: 'ABC',
            tittle: 'Image tittle',
            url: 'https://localhost',
        }]

        useFetchGifs.mockReturnValue({
            data: gifs, 
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category }/>);
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'p' ).exists() ).toBe( false );
        expect( wrapper.find( 'GifGridItem').length ).toBe( gifs.length );
    }) 
        
})