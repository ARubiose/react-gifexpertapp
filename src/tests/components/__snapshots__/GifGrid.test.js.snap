// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid should correctly show the component 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing GifGrid should show items when images are loaded from useFetchGifs 1`] = `
<Fragment>
  <h3>
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      tittle="Image tittle"
      url="https://localhost"
    />
  </div>
</Fragment>
`;
